source/bsd/kqueue_event_loop.c

```diff
@@ -651,13 +666,19 @@ static bool s_is_event_thread(struct aws_event_loop *event_loop) {
     return aws_thread_current_thread_id() == aws_thread_get_id(&impl->thread);
 }
 
-/* Called from thread.
- * Takes tasks from tasks_to_schedule and adds them to the scheduler. */
-static void s_process_tasks_to_schedule(struct aws_event_loop *event_loop, struct aws_linked_list *tasks_to_schedule) {
+static void s_process_cross_thread_data(struct aws_event_loop *event_loop) {
     struct kqueue_loop *impl = event_loop->impl_data;
+    aws_mutex_lock(&impl->cross_thread_data.mutex);
+    impl->cross_thread_data.thread_signaled = false;
 
-    while (!aws_linked_list_empty(tasks_to_schedule)) {
-        struct aws_linked_list_node *node = aws_linked_list_pop_front(tasks_to_schedule);
+    bool initiate_stop = (impl->cross_thread_data.state == EVENT_THREAD_STATE_STOPPING) &&
```

Comment:
/* copy thread state to event loop thread, will induce shutdown */

I had to go back to the header to figure out why this was safe :)